### PR TITLE
[CARBONDATA-2566] Optimize CarbonReaderExample

### PR DESCRIPTION
**Pull request by xubo245**
Optimize CarbonReaderExample
1.Add  different data type, including date and timestamp
2. update the doc
3.invoke the   
 Schema schema = CarbonSchemaReader
                .readSchemaInSchemaFile(dataFiles[0].getAbsolutePath())
                .asOriginOrder();

 - [x] Any interfaces changed?
 No
 - [x] Any backward compatibility impacted?
 No
 - [x] Document update required?
Yes, updated
 - [x] Testing done
       update the example
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
No

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6204/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5042/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5176/

**Comment by sgururajshetty** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6208/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5177/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5046/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5178/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6212/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5050/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5181/

**Comment by xubo245** [comment]
retest sdv please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5186/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5189/

File: docs/sdk-guide.md [hunk 1]
```diff
@@ -408,17 +408,30 @@ External client can make use of this reader to read CarbonData files without Car
...
     // 2. Read data
+    long day = 24L * 3600 * 1000;
     int i = 0;
     while (reader.hasNext()) {
-      Object[] row = (Object[]) reader.readNextRow();
-      System.out.println(row[0] + "\t" + row[1]);
-      i++;
+        Object[] row = (Object[]) reader.readNextRow();
+        System.out.println(
+            i + ":\t" +
```
Comment (jackylk) [inline docs/sdk-guide.md]: change to use `String.format`

File: examples/spark2/src/main/java/org/apache/carbondata/examples/sdk/CarbonReaderExample.java [hunk 2]
```diff
@@ -39,36 +42,89 @@ public static void main(String[] args) {
...
+                .build();
 
             System.out.println("\nData:");
+            long day = 24L * 3600 * 1000;
+            int i = 0;
             while (reader.hasNext()) {
                 Object[] row = (Object[]) reader.readNextRow();
-                System.out.println(row[0] + " " + row[1]);
+                System.out.println(
+                    i + ":\t" +
```
Comment (jackylk) [inline examples/spark2/src/main/java/org/apache/carbondata/examples/sdk/CarbonReaderExample.java]: change to use `String.format`

Comment (xubo245) [reply examples/spark2/src/main/java/org/apache/carbondata/examples/sdk/CarbonReaderExample.java]: ok,done

File: docs/sdk-guide.md (hunk 1 above)
Comment (xubo245) [reply docs/sdk-guide.md]: ok, done

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6217/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5192/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5055/

**Comment by xubo245** [comment]
@jacky CI pass, please check it

File: docs/sdk-guide.md [hunk 3]
```diff
@@ -408,17 +408,22 @@ External client can make use of this reader to read CarbonData files without Car
     String path = "./testWriteFiles";
     CarbonReader reader = CarbonReader
         .builder(path, "_temp")
-        .projection(new String[]{"name", "age"})
+        .projection(new String[]{"stringField", "shortField", "intField", "longField",
```
Comment (sounakr) [inline docs/sdk-guide.md]: All an example for ProjectAllColumns

Comment (xubo245) [reply docs/sdk-guide.md]: ok,done

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6223/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5195/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5061/

**Comment by sounakr** [comment]
LGTM

**Pull request closed without merging**